From c65db4f189bc00007b13304f657192fcd80c0416 Mon Sep 17 00:00:00 2001
From: Sagar Sahu <sahusag@codeaurora.org>
Date: Wed, 3 Jul 2019 19:02:40 +0530
Subject: [PATCH 2/4] tensorflow: Remove android dependecy and fix sdk version

Remove Android falg and sdk version to compile and enable NNAPI
delegate feature on Linux
---
 tensorflow/lite/nnapi/NeuralNetworksShim.h    | 2 --
 tensorflow/lite/nnapi/nnapi_implementation.cc | 2 +-
 2 files changed, 1 insertion(+), 3 deletions(-)
 mode change 100644 => 100755 tensorflow/lite/nnapi/NeuralNetworksShim.h
 mode change 100644 => 100755 tensorflow/lite/nnapi/nnapi_implementation.cc

diff --git a/tensorflow/lite/nnapi/NeuralNetworksShim.h b/tensorflow/lite/nnapi/NeuralNetworksShim.h
old mode 100644
new mode 100755
index c48528f..f106f17
--- a/tensorflow/lite/nnapi/NeuralNetworksShim.h
+++ b/tensorflow/lite/nnapi/NeuralNetworksShim.h
@@ -42,12 +42,10 @@ inline void* loadLibrary(const char* name) {
   // TODO: change RTLD_LOCAL? Assumes there can be multiple instances of nn
   // api RT
   void* handle = nullptr;
-#ifdef __ANDROID__
   handle = dlopen(name, RTLD_LAZY | RTLD_LOCAL);
   if (handle == nullptr) {
     NNAPI_LOG("nnapi error: unable to open library %s", name);
   }
-#endif
   return handle;
 }

diff --git a/tensorflow/lite/nnapi/nnapi_implementation.cc b/tensorflow/lite/nnapi/nnapi_implementation.cc
old mode 100644
new mode 100755
index 6b3de3c..ee5ff45
--- a/tensorflow/lite/nnapi/nnapi_implementation.cc
+++ b/tensorflow/lite/nnapi/nnapi_implementation.cc
@@ -101,7 +101,7 @@ int ASharedMemory_create(const char* name, size_t size) {

 const NnApi LoadNnApi() {
   NnApi nnapi = {};
-  nnapi.android_sdk_version = 0;
+  nnapi.android_sdk_version = 28;

 #ifdef __ANDROID__
   void* libandroid = nullptr;
--
1.9.1

